modules/client/src/addresslistVoting/internal/client/methods.ts

```diff
@@ -240,6 +254,99 @@ export class AddresslistVotingClientMethods extends ClientCore
       key: ExecuteProposalStep.DONE,
     };
   }
+  /**
+   * Prepares the installation of a token voting plugin in a given dao
+   *
+   * @param {AddresslistVotingPluginPrepareInstallationParams} params
+   * @return {*}  {AsyncGenerator<PrepareInstallationStepValue>}
+   * @memberof MultisigClientMethods
+   */
+  public async *prepareInstallation(
+    params: AddresslistVotingPluginPrepareInstallationParams,
+  ): AsyncGenerator<PrepareInstallationStepValue> {
+    const signer = this.web3.getConnectedSigner();
+    if (!signer) {
+      throw new NoSignerError();
+    } else if (!signer.provider) {
+      throw new NoProviderError();
+    }
+    const network = await signer.provider.getNetwork();
+    const networkName = network.name as SupportedNetworks;
```

Comment:
remove the typecasting of `SupportedNetworksArray` to be a `const`. Then you don't need the typecasting here.
To get the autocompletion back from removing the typecasting to `const` you switch it out a little in the `modules/client/src/client-common/interfaces/common.ts` file:
how it looks now:
```typescript
export const SupportedNetworksArray = ["mainnet", "goerli"] as const;
export type SupportedNetworks = typeof SupportedNetworksArray[number];
```
to this
```typescript
export type SupportedNetworks = "mainnet" | "goerli";
export const SupportedNetworksArray: SupportedNetworks[] = ["mainnet", "goerli"];
```

Reply:
Okay, I did  it that way to only have a place to do this, not the thing that is going to change the most so no problem in changing it to your way

Reply:
@mathewmeconry I don't get it, the change you propose will result in the exact same effective `type`.  Ethers is returning a `string` in network.name` I think we need this type casting